server/lib/manifest.py

```diff
@@ -0,0 +1,395 @@
+
+
+class Manifest:
+    """
+    Class that represents the manifest file.
+    Methods that add information to the manifest file have the form
+    add_<someProperty>
+    while methods that create chunks of the manifest have the form
+    create<someProperty>
+    """
+
+    def __init__(self, tale, user, item_ids=None):
+        """
+        Initialize the manifest document with base variables
+        :param tale: The Tale whose data is being serialized
+        :param user: The user requesting the manifest document
+        :param item_ids: An optional list of items to include in the manifest
+        """
+        self.tale = tale
+        self.user = user
+
+        self.manifest = dict()
+        self.item_ids = item_ids
+        # Create a set that represents any external data packages
+        self.datasets = set()
+
+        self.folderModel = ModelImporter.model('folder')
+        self.itemModel = ModelImporter.model('item')
+        self.userModel = ModelImporter.model('user')
+
+        self.manifest.update(self.create_context())
+        self.manifest.update(self.create_basic_attributes())
+        self.add_tale_creator()
+
+        if self.item_ids:
+            self.add_item_records()
+        else:
+            self.add_tale_records()
+        # Add any external datasets to the manifest
+        self.add_dataset_records()
+        self.add_system_files()
+
+    publishers = {
+        "DataONE":
+            {
+                "@id": "https://www.dataone.org/",
+                "@type": "Organization",
+                "legalName": "DataONE",
+                "Description": "A federated data network allowing access to science data"
+            },
+        "Globus":
+            {
+                "@id": "https://www.materialsdatafacility.org/",
+                "@type": "Organization",
+                "legalName": "Materials Data Facility",
+                "Description": "A simple way to publish, discover, and access materials datasets"
+            }
+    }
+
+    def create_basic_attributes(self):
+        """
+        Returns a portion of basic attributes in the manifest
+        :return: Basic information about the Tale
+        """
+
+        return {
+            "@id": 'https://data.wholetale.org/api/v1/tale/' + str(self.tale['_id']),
+            "createdOn": str(self.tale['created']),
+            "schema:name": self.tale['title'],
+            "schema:description": self.tale.get('description', str()),
+            "schema:category": self.tale['category'],
+            "schema:identifier": str(self.tale['_id']),
+            "schema:version": self.tale['format'],
+            "schema:image": self.tale['illustration'],
+            "aggregates": list(),
+            "Datasets": list()
+        }
+
+    def add_tale_creator(self):
+        """
+        Adds basic information about the Tale author
+        """
+
+        tale_user = self.userModel.load(self.tale['creatorId'], user=self.user)
```

Comment:
It won't work for public Tales (you want be to load model except for self.user itself). It needs to be forced instead.

Reply:
Fixed & Put a unit test in to make sure we can load the user's model